### PR TITLE
API and CSV fixes

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Updates auth API permissions to restrict access for non-admins
* Updates ValuesViewset to restore default filtering behaviour when doing object detail queries
* Filters both detail and list endpoints in permissions filter
* Adds sanitization to CSV writing

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by radinamatic** [comment]
- EXE asset installed on Windows 10 ✔️ 
- imported a CSV with 70 users ✔️ 
- 2 new classes created, learners enrolled, coaches assigned ✔️ 
- class deleted ✔️ 
- 3 groups created, learners searched for and assigned ✔️ 
- group deleted ✔️ 
- details added to 2 users  ✔️ 
- CSV exported and checked to confirm all the changes ✔️ 

No sign of UI issues and/or errors in the console, good to go! 👍🏽

**Review by radinamatic — APPROVED** [review]
Manual QA is looking good, :shipit:!

File: kolibri/core/api.py [hunk 1]
```diff
@@ -262,7 +240,9 @@ def serialize_object(self, **filter_kwargs):
         queryset = self.get_queryset()
         try:
             filter_kwargs = filter_kwargs or self._get_lookup_filter()
-            return self.serialize(queryset.filter(**filter_kwargs))[0]
+            return self.serialize(
+                self.filter_queryset(queryset.filter(**filter_kwargs))
```
Comment (jredrejo) [inline kolibri/core/api.py]: three `filter` in a line, not the most readable code I've seen :)

File: kolibri/core/auth/api.py [hunk 2]
```diff
@@ -99,9 +99,7 @@ class KolibriAuthPermissionsFilter(filters.BaseFilterBackend):
     """
 
     def filter_queryset(self, request, queryset, view):
-        if request.method == "GET" and request.resolver_match.url_name.endswith(
-            "-list"
-        ):
+        if request.method == "GET":
             # only filter down the queryset in the case of the list view being requested
```
Comment (jredrejo) [inline kolibri/core/auth/api.py]: we should remove this comment now

File: kolibri/core/auth/management/commands/bulkexportusers.py [hunk 3]
```diff
@@ -107,14 +108,7 @@ def kind_of_roles(field, obj):
...
 
-def map_output(obj):
-    mapped_obj = {}
-    for header, label in labels.items():
-        if header in output_mappings and header in obj:
-            mapped_obj[label] = output_mappings[header](obj)
-        elif header in obj:
-            mapped_obj[label] = obj[header]
-    return mapped_obj
+map_output = partial(output_mapper, labels=labels, output_mappings=output_mappings)
```
Comment (jredrejo) [inline kolibri/core/auth/management/commands/bulkexportusers.py:111]: :100: I had left a comment in csv.py suggesting this exact change :)

File: kolibri/core/auth/permissions/base.py [hunk 4]
```diff
@@ -108,7 +108,12 @@ def __init__(
         self.can_be_updated_by = can_be_updated_by
         self.can_be_deleted_by = can_be_deleted_by
         self.target_field = target_field
-        self.collection_field = collection_field
+        if collection_field == "self":
```
Comment (jredrejo) [inline kolibri/core/auth/permissions/base.py]: I'm lost here, could you add a comment explaining the purpose of these changes of `collection_field` values?

File: kolibri/core/auth/management/commands/bulkexportusers.py (hunk 3 above)
Comment (jredrejo) [inline kolibri/core/auth/management/commands/bulkexportusers.py:111]: :+1: I was thinking exactly of this when I saw your changes in csv.py :)

File: kolibri/core/auth/api.py (hunk 2 above)
Comment (jredrejo) [inline kolibri/core/auth/api.py]: these comments should be removed now

**Review by jredrejo — APPROVED** [review]
lThere's one point I don't quite  understand so it would be great if you can comment it.
For the most part, loving these changes

**Comment by rtibbles** [comment]
Have addressed all comments - as long as tests pass, this should be good to merge.

**Pull request merged**